crates/biome_js_analyze/src/assists/source/sort_jsx_props.rs

```diff
@@ -0,0 +1,178 @@
+    /// in which case they are sorted before callbacks.
+    /// Default is `ignore`.
+    ///
+    /// #### Example
+    ///
+    /// ```js
+    /// // shorthand first
+    /// <Hello active validate name="John" tel={5555555} />;
+    /// // shorthand last
+    /// <Hello name="John" tel={5555555} active validate />;
+    /// ```
+    ///
+    /// ### `multiline`
+    ///
+    /// When `first`, multiline props are sorted first, unless `shorthand` is `first`,
+    /// in which case they are sorted after shorthand props.
+    /// When `last`, multiline props are sorted last, unless `shorthand` is `last` or `callbacksLast` is `true`,
+    /// in which case they are sorted before shorthand props or callbacks.
+    /// Default is `ignore`.
+    ///
+    /// #### Example
+    ///
+    /// ```js
+    /// // multiline first
+    /// <Hello
+    ///   classes={{
+    ///     greetings: classes.greetings
+    ///   }}
+    ///   name="John"
+    ///   tel={5555555}
+    ///   active
+    ///   validate
+    /// />;
+    /// // multiline last
+    /// <Hello
+    ///   name="John"
+    ///   tel={5555555}
+    ///   active
+    ///   validate
+    ///   classes={{
+    ///     greetings: classes.greetings
+    ///   }}
+    /// />;
+    /// ```
+    ///
+    pub SortJsxProps {
+        version: "next",
+        name: "sortJsxProps",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintReact("jsx-sort-props")],
+        source_kind: RuleSourceKind::SameLogic,
+    }
+}
+
+impl Rule for SortJsxProps {
+    type Query = Ast<JsxAttributeList>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(_ctx: &RuleContext<Self>) -> Self::Signals {
+        Some(())
+    }
+
+    fn action(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<JsRuleAction> {
+        let props = ctx.query().clone();
+        let mut non_spread_props: Option<Vec<_>> = None;
+        let mut new_props = Vec::new();
+        for prop in props.clone() {
+            match prop {
+                AnyJsxAttribute::JsxAttribute(attr) => {
+                    if let Some(non_spread_props) = &mut non_spread_props {
+                        non_spread_props.push(attr);
+                    } else {
+                        non_spread_props = Some(vec![attr]);
+                    }
+                }
+                AnyJsxAttribute::JsxSpreadAttribute(_) => {
+                    if let Some(mut non_spread_props) = non_spread_props.take() {
+                        non_spread_props.sort_by(compare_props());
+                        new_props.extend(
+                            non_spread_props
+                                .into_iter()
+                                .map(AnyJsxAttribute::JsxAttribute),
+                        );
+                    }
+                    non_spread_props = None;
+                    new_props.push(prop);
+                }
```

Comment:
I am not really sure I understand this part. Why do we run a sorting every time we encounter a spread prop?

Reply:
The idea is that, given that a spread props resets the sorting order, every time we encounter a spread prop, we sort the previous non-spread props element.

Reply:
I think I understand what you mean, but the lack of documentation doesn't help to fully understand the algorithm. I'll wait once you address that

Reply:
Then, I believe the same approach we use in the organise imports would work. It makes things easier, I think.

Essentially, you create groups. Each group is delimited by a spread prop, where the spread prop is always the last one of the group. 

Each group is sorted in alphabetical order (I think natural order is better), and groups aren't sorted, they are replaced as they were found.